// Copyright (C) 2012 Nexon Korea Corporation All Rights Reserved.
//
// This work is confidential and proprietary to Nexon Korea Corporation and
// must not be used, disclosed, copied, or distributed without the prior
// consent of Nexon Korea Corporation.

// source: pacman_client_messages.proto
//         pacman_server_messages.proto

import "account_messages.proto";
import "pacman_client_messages.proto";
import "pacman_server_messages.proto";

message ClientAppMessageType {
  enum Type {
    kLoadLevel = 1001;
    kPacmanMove = 1002;
  }
}

extend ClientAppMessage {
  optional ClientAppMessageType.Type client_message_type = 1000;

  optional LoadLevel load_level = 1001;
  optional PacmanMove pacman_move = 1002;
}

message ServerAppMessageType {
  enum Type {
    kServerAppTimeMessage = 1001;
  }
}


extend ServerAppMessage {
  optional ServerAppMessageType.Type server_message_type = 1000;

  optional ServerAppTimeMessage timer = 1001;
}